### PR TITLE
fix: complete enum variants as patterns in pattern path

### LINKED ISSUE #12593: VSCode: Enum struct variant patterns auto-complete to invalid code
When auto-completing an enum struct variant, i.e. starting to type `Test::Va` and hitting enter, I get the following completion:

image

Please remove the `: ()` part from the completion.

### PR DESCRIPTION
**Pull request by yue4u**
close #12593

### CODE REVIEW

File: crates/ide-completion/src/tests/use_tree.rs [hunk 1]
```diff
@@ -165,30 +165,15 @@ fn enum_plain_qualified_use_tree() {
         r#"
 use Foo::$0
 
-enum Foo { Variant }
```
Comment (yue4u) [inline crates/ide-completion/src/tests/use_tree.rs:168]: These two was similar and overlapped

File: crates/ide-completion/src/completions/pattern.rs [hunk 2]
```diff
@@ -151,7 +151,6 @@ pub(crate) fn complete_pattern_path(
                     };
 
                     if let Some(hir::Adt::Enum(e)) = ty.as_adt() {
-                        cov_mark::hit!(enum_plain_qualified_use_tree);
```
Comment (yue4u) [inline crates/ide-completion/src/completions/pattern.rs:154]: This seems redundant as the check https://github.com/rust-lang/rust-analyzer/blob/09ac44c078dcbad6c19be0430cd9e1248010778d/crates/ide-completion/src/tests/use_tree.rs#L163 is hitted by https://github.com/rust-lang/rust-analyzer/blob/439a513efb77b956ed318cef6a4af421054d29e8/crates/ide-completion/src/completions/use_.rs#L81

File: crates/ide-completion/src/completions.rs [hunk 3]
```diff
@@ -357,6 +357,15 @@ impl Completions {
         variant: hir::Variant,
         local_name: Option<hir::Name>,
     ) {
+        if let PathCompletionCtx {
+            kind: PathKind::Pat { pat_ctx }, use_tree_parent: false, ..
```
Comment (yue4u) [inline crates/ide-completion/src/completions.rs:361]: This actually works without `use_tree_parent: false`, just for explicit and extra safety.

**Comment by Veykril** [comment]
Lgtm!
@bors delegate+

**Review by yue4u — COMMENTED** [review]
Hi @Veykril, thanks for the review! With #12636 merged I changed something in this PR that maybe you still want to take a look!

File: crates/ide-completion/src/tests/pattern.rs [hunk 4]
```diff
@@ -162,8 +162,9 @@ fn foo() {
             st Tuple
             st Unit
             ev Variant
-            bn Record            Record { field$1 }$0
-            bn Tuple             Tuple($1)$0
+            bn Record {…}        Record { field$1 }$0
+            bn Tuple(…)          Tuple($1)$0
+            bn Variant           Variant$0
```
[the remark this answers is not in the document]
Comment (Veykril) [reply crates/ide-completion/src/tests/pattern.rs:167]: These patterns are irrefutable and the variant can be irrefutably matched so its completed here again. The duplication is fine for now, I'm planning on rewriting the pattern completions a bit more to get rid of  that again in the future

**Pull request merged**